python/sdk/merlin/model.py

```diff
@@ -1525,6 +1469,82 @@ def _wait_build_complete(self, logs):
             return
         raise BuildError('Unknown', logs)
 
+    def _get_env_list(self) -> List[client.models.Environment]:
+        return EnvironmentApi(self._api_client).environments_get()
+
+    @staticmethod
+    def _get_default_target_env_name(env_list: List[client.models.Environment]) -> str:
+        target_env_name = None
+        for env in env_list:
+            if env.is_default:
+                target_env_name = env.name
+        if target_env_name is None:
+            raise ValueError("Unable to find default environment, "
+                             "pass environment_name to the method")
+        return target_env_name
+
+    @staticmethod
+    def _get_default_resource_request(env_name: str, env_list: List[client.models.Environment]) -> client.ResourceRequest:
+        resource_request = None
+        for env in env_list:
+            if env.name == env_name:
+                resource_request = ResourceRequest(
+                    env.default_resource_request.min_replica,
+                    env.default_resource_request.max_replica,
+                    env.default_resource_request.cpu_request,
+                    env.default_resource_request.memory_request,
+                )
+
+        # This case is when the default resource request is not specified in the environment config
+        if resource_request is None:
+            raise ValueError("default resource request not found in the environment config")
+
+        resource_request.validate()
+        return client.ResourceRequest(
+            resource_request.min_replica, resource_request.max_replica,
+            resource_request.cpu_request, resource_request.memory_request)
+
+    @staticmethod
+    def _get_default_autoscaling_policy(deployment_mode: str) -> client.AutoscalingPolicy:
+        # If a previous model version does not exist, we set up a default autoscaling_policy
+        if deployment_mode == DeploymentMode.RAW_DEPLOYMENT:
+            autoscaling_policy = RAW_DEPLOYMENT_DEFAULT_AUTOSCALING_POLICY
+        else:
+            autoscaling_policy = SERVERLESS_DEFAULT_AUTOSCALING_POLICY
+        return client.AutoscalingPolicy(autoscaling_policy.metrics_type.value, autoscaling_policy.target_value)
+
+    @staticmethod
+    def _add_env_vars(target_env_vars, new_env_vars):
+        if not isinstance(new_env_vars, dict):
+            raise ValueError(
+                f"env_vars should be dictionary, got: {type(new_env_vars)}")
+
+        if len(new_env_vars) > 0:
+            for name, value in new_env_vars.items():
+                target_env_vars.append(
+                    client.EnvVar(str(name), str(value)))
+        return target_env_vars
+
+    @staticmethod
+    def _create_transformer_spec(transformer: Transformer, target_env_name: str, env_list: List[client.models.Environment]) -> client.Transformer:
```

Comment:
This is actually almost the same as the original `create_transformer_spec` method - the only changes I made was make this method static and reused some other helper methods.